src/Editor/EditorSupport.php

```diff
@@ -0,0 +1,111 @@
+<?php
+/**
+ * Functionality around editor support for AMP plugin features.
+ *
+ * @since 2.1
+ *
+ * @package AmpProject\AmpWP
+ */
+
+namespace AmpProject\AmpWP\Editor;
+
+use AMP_Post_Type_Support;
+use AmpProject\AmpWP\Infrastructure\Registerable;
+use AmpProject\AmpWP\Infrastructure\Service;
+
+/**
+ * EditorSupport class.
+ *
+ * @internal
+ */
+final class EditorSupport implements Registerable, Service {
+
+	/**
+	 * The minimum version of Gutenberg supported by editor features.
+	 *
+	 * @var string
+	 */
+	const GB_MIN_VERSION = '5.4.0';
+
+	/**
+	 * The minimum version of WordPress supported by editor features.
+	 *
+	 * @var string
+	 */
+	const WP_MIN_VERSION = '5.3';
+
+	/**
+	 * Runs on instantiation.
+	 */
+	public function register() {
+		add_action( 'admin_enqueue_scripts', [ $this, 'maybe_show_notice' ], 99 );
+	}
+
+	/**
+	 * Shows a notice in the editor if the Gutenberg or WP version prevents plugin features from working.
+	 */
+	public function maybe_show_notice() {
+		$screen = get_current_screen();
+
+		if ( ! $screen ) {
+			return;
+		}
+
+		if ( ! isset( $screen->is_block_editor ) || false === $screen->is_block_editor ) {
+			return;
+		}
+
+		if ( ! in_array( get_post_type(), AMP_Post_Type_Support::get_eligible_post_types(), true ) ) {
+			return;
+		}
+
+		if ( ! $this->editor_supports_amp_block_editor_features() ) {
+			if ( current_user_can( 'manage_options' ) ) {
```

Comment:
These two conditions should probably be combined, or else the first should be a separate `if` like the previous and `return` if true.

Reply:
Updated in cbc69c8e2